tests/DCPS/Compiler/optional/CMakeLists.txt

```diff
@@ -8,7 +8,8 @@ include(opendds_testing)
 add_executable(${PROJECT_NAME}  "main.cpp")
 set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME optional_annotation)
 opendds_target_sources(${PROJECT_NAME} PUBLIC "optional.idl" OPENDDS_IDL_OPTIONS "-Lc++11")
-target_link_libraries(${PROJECT_NAME} OpenDDS::Dcps)
+target_link_libraries(${PROJECT_NAME} OpenDDS::Dcps gtest)
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_17)
```

Comment:
This and `tests/DCPS/optional` will have to be optionally included in the CMake build otherwise it will error if a compiler doesn't have C++17 support. For this test,

https://github.com/OpenDDS/OpenDDS/blob/81660e1d67b2230b029c0cbe2430ab4317129ea2/tests/DCPS/Compiler/CMakeLists.txt#L7-L10

would have to be:

```cmake
if(NOT OPENDDS_CXX_STD_YEAR LESS 2017)
  add_subdirectory(optional)
endif()
  
if(NOT OPENDDS_CXX_STD_YEAR LESS 2011)
  add_subdirectory(char_literals)
endif()